### PR TITLE
Adapt to the policy container

### PR DESCRIPTION
**Pull request by antosart**
This is a companion PR of a PR onto the html spec adding a policy container and storing CSP lists inside the policy container. The PR onto the html allows to remove the policy inheritance parts from the CSP spec, since inheritance is defined for the policy container directly.

### CODE REVIEW

**Review by mikewest — COMMENTED** [review]
This looks pretty reasonable to me, and it's a nice simplification. Let's wait until the HTML patch is further along before digging in here to any more depth.

File: index.bs [hunk 1]
```diff
@@ -1139,147 +1134,70 @@ spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
...
-      controls (`blob:` resources, or `document.write()`).
+  Given a {{Document}} (|document|), the user agent performs the following
+  steps in order to initialize |document|'s CSPs:
 
-  2.  For each |policy| in |response|'s <a for="response">CSP list</a>, insert
-      |policy| into |document|'s <a for="Document">CSP list</a>.
-
-  3.  For each |policy| in |document|'s <a for="Document">CSP list</a>:
+  1. For each |policy| in |document|'s [=document/policy-container=]'s
+     [=policy-container/csp-list=] <a for="policy container">CSP list</a>:
```
Comment (letitz) [inline index.bs:1181]: Looks like csp list is repeated here?

File: index.bs [hunk 2]
```diff
@@ -1139,147 +1134,70 @@ spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
...
     Retrieve the <a for="global object">CSP list</a> of an |object|
   </h4>
 
   To obtain |object|'s <a for="global object">CSP list</a>:
 
-  1.  If |object| is a {{Document}} return |object|'s <a for="Document">CSP list</a>.
-
-  2.  If |object| is a {{Window}} return |object|'s <a>associated `Document`</a>'s
-      <a for="Document">CSP list</a>.
+  1.  If |object| is a {{Document}} return |object|'s [=policy container/CSP list=].
```
Comment (letitz) [inline index.bs:1194]: Should this be:

```
|object|'s [=document/policy container=]'s [=policy container/CSP list=]
```

?

File: index.bs [hunk 3]
```diff
@@ -1139,147 +1134,70 @@ spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
...
 
-  3.  If |object| is a {{WorkerGlobalScope}}, return |object|'s <a for="global object">CSP list</a>.
+  2.  If |object| is a {{Window}} or a {{WorkerGlobalScope}} or a {{WorkletGlobalScope}},
+      return <a>environment settings object</a>'s [=environment settings object/policy
+      container=].
```
Comment (letitz) [inline index.bs:1198]: This returns a policy container instead of a CSP list. Should this be:

```
[=environment settings object/policy container=]'s [=policy container/CSP list=]
```

?

File: index.bs [hunk 4]
```diff
@@ -1125,11 +1121,10 @@ spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
...
 
-  1.  The {{Document}}, {{WorkerGlobalScope}}, and {{WorkletGlobalScope}} objects have a
-      `CSP list`, which holds all the <a for="/">policy</a> objects which are
-      active for a given context. This list is empty unless otherwise specified,
-      and is populated via the [[#initialize-global-object-csp]] and
-      [[#initialize-document-csp]] algorithms.
+  1.  The [=policy container=] has a <a for="policy container">CSP list</a>, which holds
+      all the <a for="/">policy</a> objects which are active for a given context. This
+      list is empty unless otherwise specified, and is populated from the <a>response</a>
+      or inherited following the rules of the [=policy container=].
```
Comment (letitz) [inline index.bs:1127]: Nit: this uses a mix of `[=these/links=]` and `<a for="those">links</a>`. Could be nice to align on one style?

**Review by letitz — COMMENTED** [review]
Drive-by review just because I'm curious.

File: index.bs (hunk 1 above)
Comment (antosart) [reply index.bs]: fixed

File: index.bs (hunk 2 above)
Comment (antosart) [reply index.bs]: Indeed, thanks!

File: index.bs (hunk 3 above)
Comment (antosart) [reply index.bs]: Right, thanks!

File: index.bs (hunk 4 above)
Comment (antosart) [reply index.bs:1127]: Yes, you are right. I was planning to go again through the links after the html change is merged, so that bikeshed checks if they are correct (I tried pointing it to my local version of the html spec, but I couldn't make it work)

File: index.bs [hunk 5]
```diff
@@ -994,25 +1017,9 @@ spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
   populates its <a for="response">CSP list</a> accordingly:
 
   <ol class="algorithm">
-    1.  Set |response|'s [=response/CSP list=] to the empty list.
```
Comment (mikewest) [inline index.bs:997]: I think we decided in the HTML PR that Fetch can drop a response's CSP list as part of this set of changes. Are you going to put up another CL removing that dependency?

File: index.bs [hunk 6]
```diff
@@ -1139,147 +1146,71 @@ spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
...
-  12. [[#should-block-navigation-request]] is called during the <a>process a
+  11. [[#should-block-navigation-request]] is called during the <a>process a
       navigate fetch</a> algorithm, and [[#should-block-navigation-response]]
       is called during the <a>process a navigate response</a> algorithm to
       apply directive's navigation checks, as well as inline checks for
       navigations to `javascript:` URLs.
 
   <h4 id="initialize-document-csp" algorithm dfn export>
-    Initialize a `Document`'s `CSP list`
+    Initialize a `Document`'s `CSPs`
```
Comment (mikewest) [inline index.bs:1186]: "CSPs" is weird. :)

How about "Initialize CSP in a Document's Policy Container" or "Initialize a Document's Policy Container's CSP List" (along with similar changes to 1190 below)?

File: index.bs [hunk 7]
```diff
@@ -131,9 +132,7 @@ spec: SHA2; urlPrefix: https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pd
 
 spec: HTML; urlPrefix: https://html.spec.whatwg.org/
   type: dfn
-    for: WorkerGlobalScope
-      text: owner set; url: concept-WorkerGlobalScope-owner-set
-    text: process the iframe attributes; url: process-the-iframe-attributes
+    text: initialise-the-document-object
```
Comment (mikewest) [inline index.bs:135]: This is strange text. I'd have expected "initialize the document object" as the text, with a URL of the dashed ID.

File: index.bs [hunk 8]
```diff
@@ -31,6 +31,7 @@ spec:html
     text: fallback base url
     text: duplicate-attribute
     text: origin; for: /
+    text: policy container; for: /
```
Comment (mikewest) [inline index.bs:34]: Presumably this will be exported from HTML, so you can remove it once it's in specref?

**Review by mikewest — COMMENTED** [review]
Quick pass, this is looking good.

File: index.bs (hunk 6 above)
Comment (antosart) [reply index.bs:1186]: What about "Initialize Document's CSP"? I would rather not cite the policy container here, since this is about checking those directives that have a global impact on the document (like sandbox), regardless of the fact that they are stored in the policy container.

File: index.bs (hunk 5 above)
Comment (antosart) [reply index.bs:997]: Right. That needs some more work, since response's CSP list is used by the CSP spec for the navigational response checks (frame-ancestors). For now, I just made the html spec not rely on them. I would refactor the CSP spec not to rely on them and then remove the reference from Fetch in later steps, if you agree.

File: index.bs (hunk 7 above)
Comment (antosart) [reply index.bs:135]: That also works, I'll change that.

File: index.bs (hunk 5 above)
Comment (antosart) [reply index.bs:1003]: Actually, I was just looking into this, and there is still one dependency for getting rid of response's CSP list, which is this sandbox check for workers https://w3c.github.io/webappsec-csp/#sandbox-response 

I'll have to think how to get rid of it.

Comment (antosart) [reply index.bs:1003]: In the mean time, I did fix the navigational response check (i.e. frame-ancestors) to use the policy container. Since it is a small thing, I put it together inside this change.

**Review by mikewest — APPROVED** [review]
Thanks! I'm happy with where this has ended up.

File: index.bs [hunk 9]
```diff
@@ -521,6 +516,38 @@ spec: INFRA; urlPrefix: https://infra.spec.whatwg.org/
...
+
+    3.  For each |policy| in |policies|:
+
+        1.  Set |policy|'s [=policy/self-origin=] to |response|'s [=response/url=]'s
+            [=url/origin=].
+
+    4.  Return |policies|.
+  </ol>
+
+  Note: When <a abstract-op lt="parse a response's Content Security Policies">parsing a response's
```
Comment (antosart) [inline index.bs:546]: note added here

**Comment by antosart** [comment]
I fixed the references now that the PR on html has been merged.

File: index.bs [hunk 10]
```diff
@@ -128,15 +128,17 @@ spec: SHA2; urlPrefix: https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pd
...
       text: "parser-inserted"
     text: origin; url: concept-origin
     text: browsing context; url: browsing-context
     text: content security policy state; url: attr-meta-http-equiv-content-security-policy
     text: create and initialize a new document object; url: initialise-the-document-object
+    text: initializing a new Document object; url: initialise-the-document-object
+    for: policy container
+      text: CSP list; url: policy-container-csp-list
+    for: document
+      text: policy container; url: concept-document-policy-container
```
Comment (annevk) [inline index.bs:141]: Are all these things not exported? We should export them...

Comment (antosart) [reply index.bs:141]: yes... it's still not clear to me what we should export and what not. I can create a PR on html and export them.

Comment (annevk) [reply index.bs:141]: It's basically public/private API for specs. We should export what other folks need to rely on and also be prepared to do some downstream outreach in the event of changes.

Comment (antosart) [reply index.bs]: Thanks, makes sense! Created PR for exporting here https://github.com/whatwg/html/pull/6648 and modified this one already.
